src/olympia/users/models.py

```diff
@@ -1059,8 +1059,12 @@ def watch_changes(old_attr=None, new_attr=None, instance=None, sender=None, **kw
     }
 
     # Log email changes.
-    if 'email' in changes and new_attr['email'] is not None:
-        log.info('Creating user history for user: %s' % instance.pk)
+    if (
+        'email' in changes
+        and new_attr['email'] is not None
+        and old_attr.get('email') is not None
```

Comment:
Technically can happen now that the initial attributes reflects initial state correctly for new instances - `email` has no default, so you start with a `UserProfile` instance with no email, and then set one, it would fire and `old_attr['email']` would be empty. A couple tests were broken because of it because then we would attempt to create a `UserHistory` with an empty email.